css/src/components/form/checkbox.scss

```diff
@@ -87,9 +90,30 @@ $checkbox-spacing: $spacer-3 !default;
 			}
 		}
 
-		&[disabled] + .checkbox-check {
-			opacity: 0.6;
-			cursor: not-allowed;
+		&[disabled] {
+			+ .checkbox-check {
+				border-color: $checkbox-unchecked-border-color;
+				opacity: 0.6;
+			}
+
+			+ .checkbox-check,
+			~ .checkbox-text {
+				cursor: not-allowed;
+			}
+		}
+
```

Comment:
we missed the mixin here I think, it's used as a selector

@include focus-visible { ... rules }

Reply:
updated